vmdb/app/models/ems_amazon.rb

```diff
@@ -49,13 +49,18 @@ def verify_credentials(auth_type=nil, options={})
     begin
       # EC2 does Lazy Connections, so call a cheap function
       with_provider_connection(options.merge(:auth_type => auth_type)) { |ec2| ec2.regions.map(&:name) }
-    rescue AWS::EC2::Errors::SignatureDoesNotMatch
+    rescue AWS::EC2::Errors::SignatureDoesNotMatch => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "SignatureMismatch - check your AWS Secret Access Key and signing method"
-    rescue AWS::EC2::Errors::AuthFailure
+    rescue AWS::EC2::Errors::AuthFailure => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "Login failed due to a bad username or password."
+    rescue AWS::Errors::MissingCredentialsError => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
+      raise MiqException::MiqHostError, "Missing credentials"
     rescue Exception => err
       $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
-      raise MiqException::MiqHostError, "Unexpected response returned from system, see log for details"
+      raise MiqException::MiqHostError, "Unexpected response returned from system: #{err.message}"
     end
```

Comment:
@himdel Yes, the details don't matter too much right now since we haven't done much yet.  I'm sure our first thought will be wrong anyway.  The idea is to get tests for the current behavior that you're fixing and allowing us to have a starting point for normalizing in some way later. 

For the purposes of *this* PR, you really want to test the re-raised exception and message.  It's perfectly fine that we will delete/change these later, the point is to ensure we're not going to have a bug that a basic test would reveal.  

These awful looking tests can help guide us in how to rewrite the code later on.  It's very obvious that this method is doing too much so anything like the `translate_exception` idea that allows us to test one thing (I get this inbound exception, what do I want to see in the UI would be great).